pkg/security/scc/byrestrictions.go

```diff
@@ -111,3 +124,39 @@ func volumePointValue(scc *securityapi.SecurityContextConstraints) points {
 	}
 	return noPoints
 }
+
+// hasCap checks for needle in haystack.
+func hasCap(needle kapi.Capability, haystack []kapi.Capability) bool {
+	for _, c := range haystack {
+		if needle == c {
+			return true
+		}
+	}
+	return false
+}
+
+// capabilitiesPointValue returns a score based on the capabilities allowed,
+// added, or removed by the SCC. This allow us to prefer the more restrictive
+// SCC.
+func capabilitiesPointValue(scc *securityapi.SecurityContextConstraints) points {
+	capsPoints := capDefaultPoints
+	capsPoints += capAddOnePoints * points(len(scc.DefaultAddCapabilities))
+	if hasCap(kapi.CapabilityAll, scc.AllowedCapabilities) {
+		capsPoints += capAllowAllPoints
+	} else if hasCap("ALL", scc.AllowedCapabilities) {
```

Comment:
Would you recommend uppercasing that "all" capability when storing/loading the object, or just do case-insensitive compare in `hasCap`?